### PR TITLE
lammps experiment.py

### PR DESCRIPTION
**Pull request by rfhaque**

### CODE REVIEW

File: repo/lammps/package.py [hunk 1]
```diff
@@ -9,10 +9,22 @@
 
 class Lammps(BuiltinLammps):
 
+  depends_on("kokkos+openmp", when="+openmp")
```
Comment (pearce8) [inline repo/lammps/package.py:12]: Is this not in the upstreamed package?  Did ORNL not need this?

File: systems/tioga/compilers/rocm/00-rocm-551-compilers.yaml [hunk 2]
```diff
@@ -20,8 +20,8 @@ compilers:
     modules: []
     environment:
       prepend_path:
-        LD_LIBRARY_PATH: /opt/cray/pe/cce/16.0.0/cce/x86_64/lib
-    extra_rpaths: [/opt/cray/pe/cce/16.0.0/cce/x86_64/lib/, /opt/cray/pe/gcc-libs/]
+        LD_LIBRARY_PATH: "/opt/cray/pe/cce/16.0.0/cce/x86_64/lib:/opt/rocm-5.5.1/lib"
```
Comment (pearce8) [inline systems/tioga/compilers/rocm/00-rocm-551-compilers.yaml]: So this works on both the static and dynamic tioga?

**Review by pearce8 — CHANGES_REQUESTED** [review]
- [ ] Please add a dry run
- [ ] Does this work on any system other than Tioga at this point?

File: repo/lammps/package.py (hunk 1 above)
Comment (rfhaque) [reply repo/lammps/package.py:12]: I think there is a standalone openmp mode in lammps where we disable the kokkos pacakge and just enable the openmp one (~kokkkos+openmp-package) With this spack spec, we always enable kokkos with the desired backend (+kossos+openmp)
We can ask ORNL which one of `+kokkos+openmp` or `+openmp-package+openmp` is preferable.

File: systems/tioga/compilers/rocm/00-rocm-551-compilers.yaml (hunk 2 above)
Comment (rfhaque) [reply systems/tioga/compilers/rocm/00-rocm-551-compilers.yaml]: I will not work on static tioga. I'll need to add the `LD_LIBRARY_PATH`

**Comment by rfhaque** [comment]
> * [ ]  Please add a dry run
Will do that
>     * [ ]  Does this work on any system other than Tioga at this point?
The code works on tioga(openmp, rocm) and ruby(openmp)

**Review by pearce8 — APPROVED** [review]

**Pull request merged**